state/favorites/operations/toggleToFavorite.js

```diff
@@ -28,23 +29,13 @@ const toggleToFavoritesLogic = createLogic({
 
       dispatch(toggleToFavoriteSuccess(id, favorite));
 
-      if (needRefetchList) {
-        const totalItems = favoritesTotalItems(state);
-        const currentPage = favoritesCurrentPage(state);
-        const itemsPerPage = 20;
-        const totalPages = Math.ceil(totalItems / itemsPerPage);
-
-        const singleItemOnTheLastPage = totalItems % itemsPerPage === 1;
-        const notSinglePage = currentPage !== 1;
-        const isLastPage = currentPage === totalPages;
-        const previousPage = currentPage - 1;
-
-        if (notSinglePage && isLastPage && singleItemOnTheLastPage) {
-          dispatch(fetchFavorites(previousPage));
-        } else {
-          dispatch(fetchFavorites(currentPage));
-        }
-      }
+      needRefetchListCalculation({
```

Comment:
I think name of this method should describe action, like `refetchListIfNeeded`

Reply:
Or you can check if refetch needed in the operation and call this method like `refetchList`

Reply:
Remaned to refetchListIfNeeded